### PR TITLE
refactor(checkbox): implementation component

### PR DESCRIPTION
**Pull request by aesteves60**

### CODE REVIEW

File: packages/ods/src/components/checkbox/src/index.ts [hunk 1]
```diff
@@ -0,0 +1 @@
+export { OdsCheckbox } from './components/ods-checkbox/ods-checkbox';
```
Comment (dpellier) [inline packages/ods/src/components/checkbox/src/index.ts:1]: export events

File: packages/ods/src/style/_checkbox.scss [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  background: var(--ods-color-primary-000);
+  cursor: pointer;
+  width: 16px;
+  height: 16px;
+
+  &::before {
+    @include icon.ods-icon();
+
+    position: absolute;
+    transform: scale(0);
```
Comment (dpellier) [inline packages/ods/src/style/_checkbox.scss:21]: why are you using a scale transormation on icon?

File: packages/storybook/stories/components/checkbox/documentation.mdx [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+<Meta of={ CheckboxStories } name='Documentation' />
+
+<Banner of={ CheckboxStories } />
+
+# Overview
+
+Checkbox are used for a list of options where the user may make a choice by selecting multiple options, including all or none.
+
+The previous components `checkbox` and `checkbox-button` have now been included in `checkbox`
```
Comment (dpellier) [inline packages/storybook/stories/components/checkbox/documentation.mdx:16]: should be in the migration page under `Usage changes`

File: packages/ods/src/components/checkbox/src/components/ods-checkbox/ods-checkbox.tsx [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+export class OdsCheckbox {
+  private inputEl?: HTMLInputElement;
+
+  @Prop({ reflect: true }) public ariaLabel: HTMLElement['ariaLabel'] = null;
+  @Prop({ reflect: true }) public ariaLabelledby?: string;
+  @Prop({ reflect: true }) public isChecked: boolean = false;
+  @Prop({ reflect: true }) public isDisabled: boolean = false;
+  @Prop({ reflect: true }) public isIndeterminate : boolean = false;
+  @Prop({ reflect: true }) public isRequired: boolean = false;
+  @Prop({ reflect: true }) public inputId?: string;
```
Comment (dpellier) [inline packages/ods/src/components/checkbox/src/components/ods-checkbox/ods-checkbox.tsx:19]: should be before `isChecked` to follow alphabetical order

**Review by dpellier — CHANGES_REQUESTED** [review]

**Review by astagnol — CHANGES_REQUESTED** [review]
- Overview : prefer checked version (like radio)
- On Safari, need to "tab" twice to focus the component

**Comment by aesteves60** [comment]
On Safari, need to "tab" twice to focus the component

Don't have browser stack licence I can't test it

File: packages/storybook/stories/components/checkbox/documentation.mdx [hunk 5]
```diff
@@ -13,11 +13,9 @@ import { LINK_ID } from '../../zeroheight';
 
 Checkbox are used for a list of options where the user may make a choice by selecting multiple options, including all or none.
 
-The previous components `checkbox` and `checkbox-button` have now been included in `checkbox`
-
 There is no longer a `checkbox` component to make clickable Element.
```
Comment (dpellier) [inline packages/storybook/stories/components/checkbox/documentation.mdx:16]: same, into `## Usages changes` was part of previous comment :)

**Review by dpellier — CHANGES_REQUESTED** [review]

**Review by dpellier — CHANGES_REQUESTED** [review]
Radio doc is no more visible on stroybook `Form` section

**Review by dpellier — APPROVED** [review]

**Review by astagnol — APPROVED** [review]

**Review by feoche — CHANGES_REQUESTED** [review]
## Documentation
- [ ] Put "Indeterminate" example after "Checked"
- [ ] Move "Alignment" example after "InputId"

**Comment by dpellier** [comment]
@feoche I think it's best to stick as close as possible to an alphabetical order matching the prop names, as it is unopiniated.
Not everyone come on this page with the same context/expectation/vision, thus doesn't expect to find the same order.

**Review by feoche — APPROVED** [review]

**Pull request merged**